src/sentry/snuba/metrics/extraction.py

```diff
@@ -751,7 +760,7 @@ def _process_query(self) -> RuleCondition:
         parsed_query = self._parse_query(self.query)
         # An on demand metric must have at least a condition, otherwise we can just use a classic metric.
         if parsed_query is None or len(parsed_query.conditions) == 0:
-            if count_if_rule_condition is None:
+            if count_if_rule_condition is None and parsed_field.function != "apdex":
```

Comment:
This condition without explanation is a bit tough to understand. What I would do it actually make a list of functions for which we don't require conditions and early return.

Reply:
```suggestion
            if parsed_field.function in CONDITIONLESS_FUNCTIONS:
            	return None
            	
            # Logic as before          	
```

Reply:
This way you really have an `Optional[RuleCondition]`, without a return `None` you will have a valid `RuleCondition` always, since the other cases will result in an exception being thrown.

Reply:
Refactored in [f9e32c2](https://github.com/getsentry/sentry/pull/55999/commits/f9e32c244d801050f9dd9c35e2528a59c23097fc)